nvflare/private/fed/client/client_aux_runner.py

```diff
@@ -104,23 +105,27 @@ def _send_fnf_requests(self):
         sleep_time = 0.5
         while True:
             time.sleep(sleep_time)
-            if self.asked_to_stop or self.abort_signal.triggered:
+            if self.abort_signal.triggered:
                 break
 
-            with self.fnf_lock:
-                if len(self.fnf_requests) <= 0:
+            if len(self.fnf_requests) <= 0:
+                if self.asked_to_stop:
+                    break
+                else:
                     sleep_time = 1.0
                     continue
 
-                with self.engine.new_context() as fl_ctx:
-                    bulk = Shareable()
-                    bulk.set_header(ReservedHeaderKey.TOPIC, topic)
-                    bulk.set_peer_props(fl_ctx.get_all_public_props())
+            with self.engine.new_context() as fl_ctx:
+                self.log_debug(fl_ctx, "preparing builk send")
```

Comment:
message should probably be "preparing bulk send"

Reply:
Good catch.  Thanks.